frontend/packages/metal3-plugin/src/utils/host-status.ts

```diff
@@ -37,23 +49,29 @@ const getBaremetalHostStatus = (host: K8sResourceKind) => {
   };
 };
 
-export const getHostStatus = (
-  host: K8sResourceKind,
-  // machine?: MachineKind,
-  // node?: NodeKind,
-) => {
+export type HostMultiStatus = {
+  status: string;
+  title: string;
+  errorMessage?: string;
+};
+
+export const getHostStatus = ({ host, nodeMaintenance }: HostStatusProps): HostMultiStatus => {
   // TODO(jtomasek): make this more robust by including node/machine status
-  // return isStartingMaintenance(node) || getBaremetalHostStatus(host);
-  return getBaremetalHostStatus(host);
+  return (
+    isUnderMaintanance(nodeMaintenance) ||
+    isStoppingMaintanance(nodeMaintenance) ||
+    getBaremetalHostStatus(host)
+  );
 };
 
-export const getSimpleHostStatus = (
-  host: K8sResourceKind,
-  // machine?: MachineKind,
-  // node?: NodeKind,
-): string => getHostStatus(host).status;
+export const getSimpleHostStatus = (props: HostStatusProps): string => getHostStatus(props).status;
```

Comment:
It seems like this function is no longer needed due to how we gather the status.